chunkset_tpl.h

```diff
@@ -99,6 +114,9 @@ Z_INTERNAL uint8_t* CHUNKMEMSET(uint8_t *out, unsigned dist, unsigned len) {
     chunk_t chunk;
     unsigned sz = sizeof(chunk);
     if (len < sz) {
+        if (dist >= len || dist >= sizeof(chunk_t)) 
+            return CSUFFIX(chunkcopy_partial)(out, from, len);
+
```

Comment:
Like mentioned in the comments, if we use minimum of `dist` and `len` for last parameter to `chunkcopy_partial` these two cases can be combined. In "worst" case the loop is gone through only once, in best case the loop is gone through less iterations than when copying just 1 byte in each iteration.

Reply:
Did you see in my comment where I had run the benchmarks?
https://www.quick-bench.com/q/DS-sqTR1E8pqgiNyCr3nm5Mw0GQ

`chunkcopy_a`:
```
if (dist >= len || dist >= 32) 
    return chunkcopy_partial(out, from, len);
do {
    *out++ = *from++;
    --len;
} while (len != 0);
return out;
```
`chunkcopy_b`:
```
int max_len = MIN(dist, len);
do {
    out = chunkcopy_partial(out, from, max_len);
    len -= max_len;
    max_len = MIN(max_len + max_len, len);
} while (len != 0);
return out;
```
`chunkcopy_c`:
```
do {
    *out++ = *from++;
    --len;
} while (len != 0);
return out;
```

![image](https://user-images.githubusercontent.com/1364432/120900023-84bc9200-c5e7-11eb-8d2f-17090d637ddc.png)

In the use case in the QuickBench, the `dist=1` and `len=9`. It goes through the loop 4 times instead of 9. However, because of all the extra branches it is actually slower.

Reply:
https://www.quick-bench.com/q/8BJmmpeNsflPpY2iJOE9NyLuqT0
https://www.quick-bench.com/q/Xm84euCcPZpYrpBrxtmT4MHQh3U

Reply:
https://www.quick-bench.com/q/njzZhYjh-PlG0o3P4S8MvajbYhM
https://www.quick-bench.com/q/AVzYmKDsz2RqzLq99UKkwYs4LJs

Reply:
I didn't see this comment earlier. Good optimization!

Reply:
It took me a few attempts to find the sweet spot where it improves performance the most... For short lengths, it's really fast as processor basically runs just one "slow" instruction (repeat move byte). For longer lengths obviously running multiple faster instructions gives better performance.